### PR TITLE
Add elementExists test and test for React.

### PR DESCRIPTION
**Pull request by ineffyble**
Add a new function, elementExists, which is used to find frameworks via telltale DOM elements.

Add a test for React, based on the [react-detector browser extension](https://github.com/kentcdodds/react-detector).

Note: This test seems to have some false negatives. While it works on simple React sites such as [boopevan.com](http://boopevan.com), it fails to detect React on Facebook.com.

### CODE REVIEW

File: tech-audit.js [hunk 1]
```diff
@@ -268,6 +269,10 @@
     clientSideListAdd("Twitter Widgets");
   }
 
+  if (elementExists('[data-reactroot]') || elementExists('[data-reactid]')) {
+    clientSideListAdd("React");
+  }
```
Comment (ticky) [inline tech-audit.js]: Ooh, this is an approach I hadn’t considered… what versions of React will this work with?

Comment (ineffyble) [inline tech-audit.js]: I haven't tested extensively, but the browser extension claims to support all versions. I tested it in a few sites built with recent. Can do more testing if useful.

Comment (ticky) [inline tech-audit.js]: Nah, just curious!

File: tech-audit.js [hunk 2]
```diff
@@ -268,6 +269,10 @@
     clientSideListAdd("Twitter Widgets");
   }
 
+  if (elementExists('[data-reactroot]') || elementExists('[data-reactid]')) {
```
Comment (ticky) [inline tech-audit.js]: This could be simplified to a single call! QuerySelector can have multiple comma-separated query segments.

Comment (ineffyble) [inline tech-audit.js]: Cool, done, tested, also moved it to the **Plugins and frameworks** section.

**Pull request merged**